### ISSUE TITLE
IMessageSink not working with assembly fixtures

### ISSUE THREAD

**Issue by siewers** [JDCain/Xunit.Extensions.AssemblyFixture#10]
Using the `IMessageSink` in fixtures doesn't seem to emit any output.

```cs
public class MyTests : IAssemblyFixture<CleanupFixture>
{
    public void Foo()
    {
        Assert.True(true);
    }
}


public class CleanupFixture : IAsyncLifetime
{
    private readonly IMessageSink _messageSink;
 
    public CleanupFixture(IMessageSink messageSink)
    {
        _messageSink = messageSink;
        messageSink.OnMessage(new DiagnosticMessage($"CONSTRUCTOR"));
    }

    public Task InitializeAsync()
    {
        _messageSink.OnMessage(new DiagnosticMessage($"INITIALIZE"));
    }
    
    public ValueTask DisposeAsync()
    {
        _messageSink.OnMessage(new DiagnosticMessage($"DISPOSE"));
    }
}
```

If I change the assembly fixture to a class fixture, it works as expected, but it defeats the purpose of assembly fixture.

NOTE: To get this output, you will have to set `diagnosticMessages` to `true` in the `xunit.runner.json` file, and make sure it's copied to the output directory. It also only seem to work from the command line with `dotnet test`.

```json
{
  "$schema": "https://xunit.net/schema/current/xunit.runner.schema.json",
  "diagnosticMessages": true 
}
```

**Comment by AlexeyRaga** [comment]
Is there any feedback on this one?
It looks to me like `IAsyncLifetime` doesn't work with Assembly Fixture at all.

When I have 
```csharp

public sealed class MyFixture : IAsyncLifetime
{
    public async Task InitializeAsync()
    {
        _uuid = Guid.NewGuid();
    }

    public async Task DisposeAsync()
    {
         Console.WriteLine(_uuid);
    }
}

public sealed class MyTests : IAssemblyFixture<MyFixture>
{
   ...
}
```

I do not see `InitialiseAsync` or `DisposeAsync` being called at all. Only the constructor of `MyFixture` is called.

Is there anything more that needs to be wired up?

**Comment by pergardebrink** [comment]
Would this solve it? I think we had the same issue: https://github.com/JDCain/Xunit.Extensions.AssemblyFixture/pull/12

**Comment by JDCain** [comment]
fixed in #12 by @pergardebrink
